app/src/main/java/org/dslul/openboard/inputmethod/latin/utils/InlineAutofillUtils.java

```diff
@@ -0,0 +1,119 @@
+import androidx.autofill.inline.v1.InlineSuggestionUi;
+
+import org.dslul.openboard.inputmethod.latin.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiresApi(api = Build.VERSION_CODES.R)
+public class InlineAutofillUtils {
+
+    private static int toPixel(int dp, Context context) {
+        return (int) TypedValue.applyDimension(COMPLEX_UNIT_DIP, dp,
+                context.getResources().getDisplayMetrics());
+    }
+
+    private static int getHeight(Context context) {
+        return context.getResources().getDimensionPixelSize(R.dimen.config_suggestions_strip_height);
+    }
+
+    public static InlineSuggestionsRequest createInlineSuggestionRequest(Context context) {
+
+        UiVersions.StylesBuilder stylesBuilder = UiVersions.newStylesBuilder();
+        @SuppressLint("RestrictedApi") InlineSuggestionUi.Style style = InlineSuggestionUi.newStyleBuilder()
+                .setSingleIconChipStyle(
+                        new ViewStyle.Builder()
+                                .setBackground(
+                                        Icon.createWithResource(context,
+                                                androidx.autofill.R.drawable.autofill_inline_suggestion_chip_background))
+                                .setPadding(0, 0, 0, 0)
+                                .build())
+                .setChipStyle(
+                        new ViewStyle.Builder()
+                                .setBackground(
+                                        Icon.createWithResource(context,
+                                                androidx.autofill.R.drawable.autofill_inline_suggestion_chip_background))
+                                .build())
+                .setStartIconStyle(new ImageViewStyle.Builder().setLayoutMargin(0, 0, 0, 0).build())
+                .setTitleStyle(
+                        new TextViewStyle.Builder()
+                                .setLayoutMargin(toPixel(4, context), 0, toPixel(4, context), 0)
+                                .setTextColor(Color.parseColor("#FF202124"))
+                                .setTextSize(12)
+                                .build())
+                .setSubtitleStyle(
+                        new TextViewStyle.Builder()
+                                .setLayoutMargin(0, 0, toPixel(4, context), 0)
+                                .setTextColor(Color.parseColor("#99202124"))
+                                .setTextSize(10)
+                                .build())
+                .setEndIconStyle(new ImageViewStyle.Builder().setLayoutMargin(0, 0, 0, 0).build())
+                .build();
+        stylesBuilder.addStyle(style);
+
+        Bundle stylesBundle = stylesBuilder.build();
+
+        final ArrayList<InlinePresentationSpec> presentationSpecs = new ArrayList<>();
+        presentationSpecs.add(new InlinePresentationSpec.Builder(new Size(100, getHeight(context)),
+                new Size(740, getHeight(context))).setStyle(stylesBundle).build());
```

Comment:
Where are these width values coming from / what are they actually used for?
It looks like widths are in pixels, while the height is in _dp_, which might cause issues on very high or low density devices (depending on what it's used for)

Reply:
The values are from the sample code provided by google. The link is in a comment before the onCreateInlineSuggestionsRequest function. They are the minimum and maximum width of each suggestion.

Reply:
I completely forgot to make the min and max size their own variables to avoid duplication. I will do that once this is resolved.

Reply:
Alright, then this is fine.
Since you used code from Google, could you add some license header with the appropriate license (and copyright notice, if there is any in the original code).

See the other files for the header, e.g. for LatinIME it's
```
/*
 * Copyright (C) 2008 The Android Open Source Project
 * modified
 * SPDX-License-Identifier: Apache-2.0 AND GPL-3.0-only
 */
```

Reply:
I have added the license header.